eth/gasprice/gasprice.go

```diff
@@ -179,50 +179,83 @@ func (gpo *Oracle) SuggestPrice(ctx context.Context) (*big.Int, error) {
 	return price, nil
 }
 
-type getBlockPricesResult struct {
-	prices []*big.Int
+type results struct {
+	values []*big.Int
 	err    error
 }
 
-type transactionsByGasPrice []*types.Transaction
+type SortableTxs interface {
+	Len() int
+	Less(i, j int) bool
+	Swap(i, j int)
+	GasPrice(int) *big.Int
+	EffectiveTip(int, *big.Int) *big.Int
+}
+
+type transactionsByFeeCap []*types.Transaction
+
+func (t transactionsByFeeCap) Len() int                                { return len(t) }
+func (t transactionsByFeeCap) Swap(i, j int)                           { t[i], t[j] = t[j], t[i] }
+func (t transactionsByFeeCap) Less(i, j int) bool                      { return t[i].FeeCapCmp(t[j]) < 0 }
+func (t transactionsByFeeCap) GasPrice(i int) *big.Int                 { return t[i].GasPrice() }
+func (t transactionsByFeeCap) EffectiveTip(i int, b *big.Int) *big.Int { return nil }
+
+type transactionsByTip []*types.Transaction
 
-func (t transactionsByGasPrice) Len() int           { return len(t) }
-func (t transactionsByGasPrice) Swap(i, j int)      { t[i], t[j] = t[j], t[i] }
-func (t transactionsByGasPrice) Less(i, j int) bool { return t[i].FeeCapCmp(t[j]) < 0 }
+func (t transactionsByTip) Len() int                { return len(t) }
+func (t transactionsByTip) Swap(i, j int)           { t[i], t[j] = t[j], t[i] }
+func (t transactionsByTip) Less(i, j int) bool      { return t[i].TipCmp(t[j]) < 0 }
```

Comment:
Why should we sort the transaction by the tip in order to suggest a suitable tip?
I think we should sort by effective_tip.